File: features/lender/src/main/java/com/nivasafinance/features/lender/lender/service/LenderWriteService.java
package com.nivasafinance.features.lender.lender.service;

import com.nivasafinance.features.lender.lender.dto.LenderRequestData;
import com.nivasafinance.features.lender.lender.dto.LenderResponseData;
import com.nivasafinance.features.lender.lender.dto.UpdateLenderRequest;

import java.util.UUID;

public interface LenderWriteService {
    LenderResponseData create(LenderRequestData lenderData);
    LenderResponseData update(UUID id, UpdateLenderRequest request);
    void delete(UUID id);
    LenderResponseData activateDeactivateLender(UUID id);
}

File: features/lender/src/main/java/com/nivasafinance/features/lender/lender/service/LenderWriteService.kt
package com.nivasafinance.features.lender.lender.service

import com.nivasafinance.features.lender.lender.dto.LenderRequestData
import com.nivasafinance.features.lender.lender.dto.LenderResponseData
import java.util.UUID

interface LenderWriteService {
    fun create(lenderData: LenderRequestData): LenderResponseData
    fun update(id: UUID, lenderData: LenderRequestData): LenderResponseData
    fun delete(id: UUID)
}
